From dd6f36cb80b15dd4dedf15402c8d2f8bdf22a33a Mon Sep 17 00:00:00 2001
From: Thu Nguyen <thu@os.amperecomputing.com>
Date: Tue, 22 Oct 2024 14:23:24 +0000
Subject: [PATCH 02/14] platform-mc: Correct the discovered terminus condition

When the terminus is discovered, the terminus shared pointer will be
added to `termini` list. So `termini[tid]` will not be `nullptr`.
Correct the discovered terminus condition.

Signed-off-by: Thu Nguyen <thu@os.amperecomputing.com>
Change-Id: I070e927d34a49d0fdbc02fcd9dbbcd4be2f897a7
---
 platform-mc/event_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform-mc/event_manager.cpp b/platform-mc/event_manager.cpp
index a784cdd..139438b 100644
--- a/platform-mc/event_manager.cpp
+++ b/platform-mc/event_manager.cpp
@@ -377,7 +377,7 @@ int EventManager::processCperEvent(pldm_tid_t tid, uint16_t eventId,
         return rc;
     }
 
-    if (termini.contains(tid) && !termini[tid])
+    if (termini.contains(tid) && termini[tid])
     {
         auto tmp = termini[tid]->getTerminusName();
         if (tmp && !tmp.value().empty())
-- 
2.44.2

